springdb/frontend/src/pages/DashboardApp.js

```diff
@@ -15,55 +16,85 @@ import {
   AppConversionRates
 } from '../sections/@dashboard/app';
 
+import CytoCard from '../sections/@dashboard/app/CytoCard';
+import AEList from '../sections/@dashboard/products/AEList';
+import dummyNodes from '../utils/dummyNodes';
+
 // ----------------------------------------------------------------------
 
-export default function DashboardApp() {
-  return (
-    <Page title="Dashboard | Minimal-UI">
-      <Container maxWidth="xl">
-        <Box sx={{ pb: 5 }}>
-          <Typography variant="h4">Hi, Welcome back</Typography>
-        </Box>
-        <Grid container spacing={3}>
-          <Grid item xs={12} sm={6} md={3}>
-            <AppWeeklySales />
-          </Grid>
-          <Grid item xs={12} sm={6} md={3}>
-            <AppNewUsers />
-          </Grid>
-          <Grid item xs={12} sm={6} md={3}>
-            <AppItemOrders />
-          </Grid>
-          <Grid item xs={12} sm={6} md={3}>
-            <AppBugReports />
-          </Grid>
+export default class DashboardApp extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
```

Comment:
This pushes state up, but perhaps it could be pushed up less, for example, with a component that encapsulates both the list and card.  That way state is only shared among the things that really need to access them.

Reply:
Yes, it's a good point. I think the answer on how high up we pull the state  will depend on how many other components use the same `graphNodes` data -- i.e. are the AE List and AE Card the only entities on the page that use data about the graph, or are there other components (e.g. a Target List or a DataSet picker) that use the same `graphNodes` data?